src/pegen/grammar_parser.py

```diff
@@ -248,21 +236,16 @@ def named_item(self) -> Optional[NamedItem]:
         if (
             (name := self.name())
             and (annotation := self.annotation())
-            and (literal := self.expect("="))
-            and (cut := True)
+            and (self.expect("="))
+            and (True)
```

Comment:
This looks wrong to me. Updating cut has side effects (l 268) and they are not taken into account.

Reply:
oh yeah, I didn't handle cut properly